pkg/api/getambassador.io/v3alpha1/crd_tracingservice.go

```diff
@@ -53,11 +91,14 @@ type TracingServiceSpec struct {
 	// +kubebuilder:validation:Required
 	Driver string `json:"driver,omitempty"`
 	// +kubebuilder:validation:Required
-	Service    string         `json:"service,omitempty"`
-	Sampling   *TraceSampling `json:"sampling,omitempty"`
-	TagHeaders []string       `json:"tag_headers,omitempty"`
-	Config     *TraceConfig   `json:"config,omitempty"`
-	StatsName  string         `json:"stats_name,omitempty"`
+	Service  string         `json:"service,omitempty"`
+	Sampling *TraceSampling `json:"sampling,omitempty"`
+	// tag_headers is deprecated. Use custom_tags instead.
```

Comment:
It looks kubebuilder has no notion of deprecation validation so let's at least take advantage of Golangs doc comments for deprecated.

https://stackoverflow.com/questions/7849663/how-do-you-mark-code-as-deprecated-in-go

Reply:
This one is a little bit tricky. The comments endup in the CRD specs. The Deprecated convention in Go is for the struct fields, not the json fields. I am not sure if this is worth it. If so, can you write down here the expected comment string?

Reply:
Correct, the reason I suggested the Go Doc comment is that these objects can be used in the future where we have a transformation engine that is written in Go. This way if objects are being instantiated via Go code the IDE is going to yell you.

However, I was reading through the docs this morning and it looks like there is a deprecated special marker. It says on-type so it might not work on fields but ideally it can be applied to a field too.
https://book.kubebuilder.io/reference/markers/crd.html

Reply:
`+kubebuilder:deprecatedversion` doesn't seem to do anything on fields. I am gonna add the Go deprecation comment and use the json fields in the description since those are visible both in the CRD schema and the Go code.